airflow/models/dagrun.py

```diff
@@ -126,6 +130,10 @@ def __init__(
     ):
         self.dag_id = dag_id
         self.run_id = run_id
+        if queued_at is self.__NO_VALUE:
+            self.queued_at = timezone.utcnow() if state == State.QUEUED else None
+        else:
+            self.queued_at = queued_at
```

Comment:
We need to do this after `self.state = state` (current L141), and `def set_state` should also set queued_at when state is being _changed_ to queued.